cmd/config.go

```diff
@@ -196,64 +199,85 @@ func (cp *ConfigProvider) configureMeters(conf config) error {
 }
 
 func (cp *ConfigProvider) configureChargers(conf config) error {
+	var mu sync.Mutex
+	g, _ := errgroup.WithContext(context.Background())
+
 	cp.chargers = make(map[string]api.Charger)
 	for id, cc := range conf.Chargers {
 		if cc.Name == "" {
 			return fmt.Errorf("cannot create %s charger: missing name", humanize.Ordinal(id+1))
 		}
 
-		c, err := charger.NewFromConfig(cc.Type, cc.Other)
-		if err != nil {
-			err = fmt.Errorf("cannot create charger '%s': %w", cc.Name, err)
-			return err
-		}
+		cc := cc
 
-		if _, exists := cp.chargers[cc.Name]; exists {
-			return fmt.Errorf("duplicate charger name: %s already defined and must be unique", cc.Name)
-		}
+		g.Go(func() error {
+			c, err := charger.NewFromConfig(cc.Type, cc.Other)
+			if err != nil {
+				return fmt.Errorf("cannot create charger '%s': %w", cc.Name, err)
+			}
+
+			mu.Lock()
+			defer mu.Unlock()
 
-		cp.chargers[cc.Name] = c
+			if _, exists := cp.chargers[cc.Name]; exists {
+				return fmt.Errorf("duplicate charger name: %s already defined and must be unique", cc.Name)
+			}
+
+			cp.chargers[cc.Name] = c
+			return nil
+		})
 	}
 
-	return nil
+	return g.Wait()
 }
 
 func (cp *ConfigProvider) configureVehicles(conf config) error {
+	var mu sync.Mutex
+	g, _ := errgroup.WithContext(context.Background())
+
 	cp.vehicles = make(map[string]api.Vehicle)
 	for id, cc := range conf.Vehicles {
 		if cc.Name == "" {
 			return fmt.Errorf("cannot create %s vehicle: missing name", humanize.Ordinal(id+1))
 		}
 
-		// ensure vehicle config has title
-		var ccWithTitle struct {
-			Title string
-			Other map[string]interface{} `mapstructure:",remain"`
-		}
+		cc := cc
 
-		if err := util.DecodeOther(cc.Other, &ccWithTitle); err != nil {
-			return err
-		}
+		g.Go(func() error {
+			// ensure vehicle config has title
+			var ccWithTitle struct {
+				Title string
+				Other map[string]interface{} `mapstructure:",remain"`
+			}
 
-		if ccWithTitle.Title == "" {
-			//lint:ignore SA1019 as Title is safe on ascii
-			cc.Other["title"] = strings.Title(cc.Name)
-		}
+			if err := util.DecodeOther(cc.Other, &ccWithTitle); err != nil {
+				return err
+			}
 
-		v, err := vehicle.NewFromConfig(cc.Type, cc.Other)
-		if err != nil {
-			// wrap any created errors to prevent fatals
-			v, _ = wrapper.New(v, err)
-		}
+			if ccWithTitle.Title == "" {
+				//lint:ignore SA1019 as Title is safe on ascii
+				cc.Other["title"] = strings.Title(cc.Name)
+			}
 
-		if _, exists := cp.vehicles[cc.Name]; exists {
-			return fmt.Errorf("duplicate vehicle name: %s already defined and must be unique", cc.Name)
-		}
+			v, err := vehicle.NewFromConfig(cc.Type, cc.Other)
+			if err != nil {
+				// wrap any created errors to prevent fatals
+				v, _ = wrapper.New(v, err)
+			}
+
+			mu.Lock()
+			defer mu.Unlock()
 
-		cp.vehicles[cc.Name] = v
+			if _, exists := cp.vehicles[cc.Name]; exists {
```

Comment:
Kann das ggf. doppelte Anlegen von Auto's zu API Problemen führen? Da es in Zeile 262 schon angelegt und instanziert wird, bevor es dann geprüft wird.

Positiv sehe ich, dass der Mutex so spät wie möglich gelockt wird. Allerdings sollte man ggf. den Absatz 262-269 hinter die xists prüfung verschieben.